src/docs/howto-listing/bookings-with-buffer/index.md

```diff
@@ -0,0 +1,411 @@
++
++  const bufferedParams = {
++    ...orderParams,
++    bookingEnd: bufferEnd,
++    bookingDisplayEnd: orderParams.bookingEnd,
++    bookingDisplayStart: orderParams.bookingStart,
++  };
+
+   const bookingData = {
+     startDate: orderParams.bookingStart,
+-    endDate: orderParams.bookingEnd,
++    endDate: bufferedParams.bookingEnd,
++    displayEnd: bufferedParams.bookingDisplayEnd,
+   };
+
+   const bodyParams = isTransition
+     ? {
+         id: transactionId,
+         transition,
+-        params: orderParams,
++        params: bufferedParams,
+       }
+     : {
+         processAlias: config.bookingProcessAlias,
+         transition,
+-        params: orderParams,
++        params: bufferedParams,
+       };
+```
+
+We also need to modify line item calculation. By default, the
+transaction's price is calculated based on the booking's start and end
+moments, however in this case we want to use the display end attribute
+for the price calculation.
+
+```shell
+└── server
+    └── api-util
+        └── lineItems.js
+```
+
+We have already passed _displayEnd_ in _bookingData_, so we just need to
+add handling for it in _lineItems.js_. Since the function is used both
+with and without display end time, e.g. when calling
+_api/transaction-line-items_, we need to accommodate both use cases.
+
+```diff
+exports.transactionLineItems = (listing, bookingData) => {
+  const unitPrice = listing.attributes.price;
+- const { startDate, endDate } = bookingData;
++ const { startDate, endDate, displayEnd } = bookingData;
+ ...
+  const booking = {
+    code: bookingUnitType,
+    unitPrice,
+-   quantity: calculateQuantityFromHours(startDate, endDate),
++   quantity: calculateQuantityFromHours(startDate, (displayEnd || endDate)),
+    includeFor: ['customer', 'provider'],
+  };
+```
+
+### Set getStartHours to return correct available start times
+
+Finally, we need to make sure that the start time slots only show valid
+start times including the buffer – if someone has booked 6pm to 7pm,
+then another customer cannot book 5pm to 6pm because there is not enough
+availability for both the time slot and the buffer.
+
+```shell
+└── src
+    └── util
+        └── dates.js
+```
+
+First, add a constant for the buffer time and a full hour.
+
+```diff
++ export const bufferMinutes = 15;
++ const hourMinutes = 60;
+```
+
+<extrainfo title="Also use bufferMinutes constant on CheckoutPage.duck.js">
+It makes sense to also use bufferMinutes constant in CheckoutPage.duck.js, so you don't have to rely on hard-coded values.
+
+```shell
+└── src
+    └── containers
+        └── CheckoutPage
+            └── CheckoutPage.duck.js
+```
+
+```diff
+  import moment from 'moment';
++ import { bufferMinutes } from '../../util/dates';
+
+  ...
+-   const bufferEnd = moment(orderParams.bookingEnd).add(15, 'minutes').toDate();
++   const bufferEnd = moment(orderParams.bookingEnd).add(bufferMinutes, 'minutes').toDate();
+```
```

Comment:
How about:

CheckoutPage.duck.js:
```diff
-  import moment from 'moment';
+ import { addBuffer } from '../../util/dates';

  ...
-   const bufferEnd = moment(orderParams.bookingEnd).add(15, 'minutes').toDate();
+   const bufferEnd = addBuffer(orderParams.bookingEnd);
```
src/util/dates.js:
```diff
export const addBuffer = (date) => moment(date).add(bufferMinutes, 'minutes').toDate();
```

---
reasoning: we have tried to keep the Moment library in as few places as possible - and dates-util is the best place to work with moment-related helpers.

Reply:
Great point! Will make this change 👍